src/Engine/ProtoCore/Core.cs

```diff
@@ -1207,9 +1208,30 @@ public List<int> GetAncestorBlockIdsOfBlock(int blockId)
             return ancestors;
         }
 
+        //public int GetCurrentBlockId()
+        //{
+        //    int constructBlockId = Rmem.CurrentConstructBlockId;
+        //    if (constructBlockId == Constants.kInvalidIndex)
+        //        return __TempCoreHostForRefactoring.DebugProps.CurrentBlockId;
+
+        //    CodeBlock constructBlock = ProtoCore.Utils.CoreUtils.GetCodeBlock(CodeBlockList, constructBlockId);
+        //    while (null != constructBlock && constructBlock.blockType == CodeBlockType.kConstruct)
+        //    {
+        //        constructBlock = constructBlock.parent;
+        //    }
+
+        //    if (null != constructBlock)
+        //        constructBlockId = constructBlock.codeBlockId;
+
+        //    if (constructBlockId != __TempCoreHostForRefactoring.DebugProps.CurrentBlockId)
+        //        return __TempCoreHostForRefactoring.DebugProps.CurrentBlockId;
+        //    else
+        //        return Rmem.CurrentConstructBlockId;
+        //}
+
         public int GetCurrentBlockId()
         {
-            int constructBlockId = Rmem.CurrentConstructBlockId;
+            int constructBlockId = __TempCoreHostForRefactoring.RuntimeMemory.CurrentConstructBlockId;
```

Comment:
Its going to be removed in the last PR. This is a place holder to make it easier to swap __TempCoreHostForRefactoring with RuntimeCore